src/lib/tls/tls13_pqc/hybrid_public_key.h

```diff
@@ -93,6 +93,16 @@ class BOTAN_TEST_API Hybrid_KEM_PrivateKey final : public Private_Key,
    public:
       Hybrid_KEM_PrivateKey(std::vector<std::unique_ptr<Private_Key>> private_keys);
 
+      /* Work around a bug/oddity in clang-cl on Windows - the default move assignment operator
+       * generates multiple calls to Public_Key(Public_Key&&) as a technically-valid-but-dangerous
+       * side-effect of the class having the dllexport attribute. In the cast of the Public_Key
+       * class, this isn't actually a problem since its just an interface and has no data members,
+       * but Clang still raises a warning for it.
+       *
+       * Hybrid_KEM_PrivateKey has an explicit move assignment operator to silence the warning.
+      */
+      Hybrid_KEM_PrivateKey &operator=(Hybrid_KEM_PrivateKey&& other);
```

Comment:
Perhaps adhere to the rule-of-five then, and also add a copy c'tor, copy assignment and destructor that should all be 'default'. It seems that adding a non-default move c'tor would also make sense give the described oddity, no?

Reply:
I've added explicit default/delete for the other move/copy assignment operators and c'tors for completeness, not inclined to add a custom move c'tor since its just a workaround for a compiler warning and not an actual issue (the warning doesn't get emited for the default move c'tors).

GCC is supposedly a bit more pragmatic here and won't emit the warning for the multiply-moved-trivially-movable-base-class case, so Clang might follow suit in the future...